internal/engine/wazevo/backend/isa/amd64/machine.go

```diff
@@ -179,6 +184,25 @@ func (m *machine) lowerAluRmiROp(si *ssa.Instruction, op aluRmiROpcode) {
 	m.insert(mov2)
 }
 
+func (m *machine) getOperand_Mem_Imm32_Reg(def *backend.SSAValueDefinition) (op operand) {
+	if def.IsFromBlockParam() {
+		return newOperandReg(def.BlkParamVReg)
+	}
+
+	instr := def.Instr
+	switch instr.Opcode() {
+	case ssa.OpcodeLoad:
```

Comment:
just simply matching on opcode is troublesome. look at arm64 carefully and you will know why